### PR TITLE
fit for color + zeropoint using x and y uncertainties

### PR DESCRIPTION
**Pull request by saarahhall**
This tiny PR utilizes `scipy.odr` inside `ZPWithColorTermCalculator` to fit a line using the x and y uncertainties in the color vs. differential mag plot. 

Room for improvement: for the ODR to work, we can't use data with 0 uncertainty (in x or y). In quite a few lines of code, I added some code that removes those points with 0 uncertainty. This could certainly be compressed. But also, 0 uncertainty in x would only happen if the reference catalog reports 0 uncertainty for both magnitudes of interest, which should be a red flag anyways. So, I feel this cut belongs somewhere else... in a reference catalog purifier? But one that everyone should use... let me know your thoughts @virajkaram @robertdstein

### CODE REVIEW

File: mirar/processors/photcal/zp_calculator.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+                )
+
+                if num_stars > self.num_stars_threshold:
+                    break
+
+            check = [np.isnan(x) for x in [zp_mean, zp_med, zp_std]]
+            if np.sum(check) > 0:
+                err = (
+                    f"Error with nan when calculating sigma stats: \n "
+                    f"mean: {zp_mean}, median: {zp_med}, std: {zp_std}"
```
Comment (robertdstein) [inline mirar/processors/photcal/zp_calculator.py:95]: pit some limits on this e.g `{zp_mean:.2f}`

File: mirar/processors/photcal/zp_calculator.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+            for zpvals in zeropoints:
+                image[f"ZP_{zpvals['diameter']}"] = zpvals["zp_mean"]
+                image[f"ZP_{zpvals['diameter']}_std"] = zpvals["zp_std"]
+                image[f"ZP_{zpvals['diameter']}_nstars"] = zpvals["nstars"]
+
+        return image
+
+
+def line_model(data, slope, intercept):
```
Comment (robertdstein) [inline mirar/processors/photcal/zp_calculator.py:123]: returns float

File: mirar/processors/photcal/zp_calculator.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    """linear model to hand to scipy curve_fit"""
```
Comment (robertdstein) [inline mirar/processors/photcal/zp_calculator.py:124]: describe arguments

File: mirar/zeropoints/zp_calculators.py
Comment (robertdstein) [inline mirar/zeropoints/zp_calculators.py:1]: why is this empty?

**Review by robertdstein — CHANGES_REQUESTED** [review]

**Review by robertdstein — APPROVED** [review]

File: mirar/processors/photcal/zp_calculator/color_term_zp_calculator.py [hunk 4]
```diff
@@ -60,9 +82,35 @@ def calculate_zeropoint(
...
-            color, zero_point = popt
-            color_err, zp_err = np.sqrt(np.diag(pcov))
+            x_err = np.sqrt(
+                matched_ref_cat[color_err_colnames[0]] ** 2
+                + matched_ref_cat[color_err_colnames[1]] ** 2
+            )
+
+            # use scipy.odr to fit a line to data with x and y uncertainties
+            ## setup: remove sources with 0 uncertainty (or else scipy.odr won't work)
+            where_zero_y = np.where(np.array(y_err) == 0)[0]
```
Comment (virajkaram) [inline mirar/processors/photcal/zp_calculator/color_term_zp_calculator.py:92]: Probably cleaner with `non_zero_mask = (np.array(y_err)==0) | (np.array(x_err)==0)`

Comment (virajkaram) [reply mirar/processors/photcal/zp_calculator/color_term_zp_calculator.py:92]: And then `x_err = x_err[non_zero_mask]` and so on, will have to convert all lists to numpy arrays

**Review by virajkaram — COMMENTED** [review]
minor change

**Pull request merged**